open_feature_contrib/providers/flagd/provider.py

```diff
@@ -0,0 +1,186 @@
+        :param schema: the schema for the transport protocol, e.g. 'http', 'https'
+        :param host: the host to make requests to
+        :param port: the port the flagd service is available on
+        :param timeout: the maximum to wait before a request times out
+        """
+        self.provider_name = name
+        self.schema = schema
+        self.host = host
+        self.port = port
+        self.timeout = timeout
+
+        self.url_factory = WebApiUrlFactory(self.schema, self.host, self.port)
+
+    def get_metadata(self):
+        """Returns provider metadata"""
+        return {
+            "name": self.get_name(),
+            "schema": self.schema,
+            "host": self.host,
+            "port": self.port,
+            "timeout": self.timeout,
+        }
+
+    def get_name(self) -> str:
+        """Returns provider name"""
+        return self.provider_name
+
+    def get_boolean_details(
+        self,
+        key: str,
+        default_value: bool,
+        evaluation_context: EvaluationContext = None,
+    ):
+        return self.__resolve(key, FlagType.BOOLEAN, default_value, evaluation_context)
+
+    def get_string_details(
+        self,
+        key: str,
+        default_value: str,
+        evaluation_context: EvaluationContext = None,
+    ):
+        return self.__resolve(key, FlagType.STRING, default_value, evaluation_context)
+
+    def get_float_details(
+        self,
+        key: str,
+        default_value: Number,
+        evaluation_context: EvaluationContext = None,
+    ):
+        return self.__resolve(key, FlagType.FLOAT, default_value, evaluation_context)
+
+    def get_int_details(
+        self,
+        key: str,
+        default_value: Number,
+        evaluation_context: EvaluationContext = None,
+    ):
+        return self.__resolve(key, FlagType.INTEGER, default_value, evaluation_context)
+
+    def get_object_details(
+        self,
+        key: str,
+        default_value: dict,
+        evaluation_context: EvaluationContext = None,
+    ):
+        return self.__resolve(key, FlagType.OBJECT, default_value, evaluation_context)
+
+    def __resolve(
+        self,
+        flag_key: str,
+        flag_type: FlagType,
+        default_value: typing.Any,
+        evaluation_context: EvaluationContext,
+    ):
+        """
+        This method is equivalent to:
+        curl -X POST http://localhost:8013/{path} \
+             -H "Content-Type: application/json" \
+             -d '{"flagKey": key, "context": evaluation_context}'
+        """
+
+        payload = {
+            "flagKey": flag_key,
+            "context": EvaluationContextSerializer.to_dict(evaluation_context),
+        }
+
+        try:
+            url_endpoint = self.url_factory.get_path_for(flag_type)
+
+            response = requests.post(
+                url=url_endpoint, timeout=self.timeout, json=payload
+            )
+
+        except Exception:
```

Comment:
We should tighten the except clause here to avoid catching unrelated exceptions.

See [here](https://stackoverflow.com/a/16511493/368861) for the exceptions that requests can raise.